smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -44,6 +44,62 @@ contract MixinTransfer is
   // This is calculated as `keyPrice * transferFeeBasisPoints / BASIS_POINTS_DEN`.
   uint public transferFeeBasisPoints;
 
+  /**
+  * @notice Allows the key owner to share their key (parent key) by
+  * transferring a portion of the remaining time to a new key (child key).
+  * @param _from The owner of the parent key
+  * @param _to The recipient of the shared key
+  * @param _tokenId the key to share
+  * @param _timeShared The amount of time shared
+  */
+  function shareKey(
+    address _from,
+    address _to,
+    uint _tokenId,
+    uint _timeShared
+  ) public
+    onlyIfAlive
+    hasValidKey(_from)
+    onlyKeyOwnerOrApproved(_tokenId)
+  {
+    require(_to != address(0), 'INVALID_ADDRESS');
+
+    Key storage fromKey = keyByOwner[_from];
+    Key storage toKey = keyByOwner[_to];
+    uint IdTo = toKey.tokenId;
+    uint time;
+    // get the remaining time for the origin key
+    uint timeRemaining = fromKey.expirationTimestamp - block.timestamp;
+    // get the transfer fee
+    uint fee = getTransferFee(_from, _timeShared);
+
+    // ensure that we don't try to share too much
+    if(_timeShared.add(fee) < timeRemaining) {
+      time = _timeShared;
```

Comment:
This results in overcrediting the other account still, right?  If I try to transfer 1 year to a friend but only have 1 day remaining, it seems I'm charged a day but they still got they full year (?)

Reply:
So, if the time I want to share + fee < the timeRemaining on my key, then `time = _timeShared`, but if the time I want to share + fee >= the timeRemaining on my key, then we expire my key and share the time remaining -fee

Make sense?

Reply:
Basically, the value of `_time` depends on whether I'm trying to share less than I have or more. In your example, "they" only got a day minus the fee.

Reply:
yup my bad